cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -729,6 +767,63 @@ class SimpleArray
     value_type * m_body = nullptr;
 }; /* end class SimpleArray */
 
+template <typename A, typename T>
+SimpleArray<uint64_t> detail::SimpleArrayMixinSort<A, T>::argsort(void)
+{
+    auto athis = static_cast<A *>(this);
+    if (athis->ndim() != 1)
+    {
+        throw std::runtime_error(Formatter() << "SimpleArray: argsort() supports only in 1D array "
+                                                " but the array is "
+                                             << athis->ndim() << " dimension");
+    }
+
+    SimpleArray<uint64_t> ret(athis->shape());
+
+    { // Return array initialization
+        uint64_t cnt = 0;
+        std::for_each(ret.begin(), ret.end(), [&cnt](uint64_t & v)
+                      { v = cnt++; });
+    }
+
+    value_type const * buf = athis->body();
+    auto cmp = [buf](uint64_t a, uint64_t b)
+    {
+        return buf[a] < buf[b];
+    };
+    std::sort(ret.begin(), ret.end(), cmp);
+    return ret;
+}
+
+template <typename A, typename T>
+template <typename I>
+A detail::SimpleArrayMixinSort<A, T>::take_along_axis(SimpleArray<I> const & indices)
+{
+    static_assert(std::is_integral_v<I>, "I must be integral type");
+    auto athis = static_cast<A *>(this);
+    if (athis->ndim() != 1)
+    {
+        throw std::runtime_error(Formatter() << "SimpleArray: take_along_axis() supports currently only in 1D array "
+                                                " but the array is "
+                                             << athis->ndim() << " dimension");
+    }
+
+    A ret(indices.shape());
+
+    auto val_iter = ret.begin();
+    for (auto idx : indices)
+    {
+        if (idx < 0 || idx > athis->shape()[0])
+        {
+            throw std::out_of_range(Formatter() << "SimpleArray: index " << idx << " in indices "
```

Comment:
The error message should writes like:

```
SimpleArray::take_along_axis() indices[i1] is i2, which is out of range of the array size v_size
```

You need 3 values:
* `i1` is the serial number of the input indices whose index is out of range
* `i2` is the index out of range
* `vsize` is the range (size)